### ISSUE TITLE
Add chainer.get_version_info()

### ISSUE THREAD

**Issue by mitmul** [chainer/chainer#4235]
Sometimes I want related package versions like this:

```
Chainer: 3.0.0b1
CUDA Version: 8.0
cuDNN Version: 5.11
CuPy: 2.0.0b1
```

This is actually the output of this code:

```
import chainer
import cupy

print('Chainer:', chainer.__version__)
print('CUDA:', chainer.cuda.cuda.runtime.runtimeGetVersion() / 1000.)
print('cuDNN:', chainer.cuda.cudnn.cudnn.getVersion() / 1000.)
print('CuPy:', cupy.__version__)
```

But I think it's better to provide more easy way for user to get those information, e.g., `chainer.get_version_info()` etc.

**Comment by kmaehashi** [comment]
:+1: Related to https://github.com/cupy/cupy/issues/756

**Comment by niboshi** [comment]
Some considerations.
* `version_info` might give an impression to be a tuple (c.f. `sys.version_info`)
* In Chainer, they're runtime configuration (unlike NumPy's `show_config()` and CuPy). We should be careful not to give an impression that they're build-time configuration.

**Comment by mitmul** [comment]
It may be useful for user support. Just asking "call `chainer.get_version_info()` before posting questions to the forum" may help us to smoothly start the support.

**Comment by niboshi** [comment]
I agree.
How about `get_runtime_info()`? (just an idea)

**Comment by niboshi** [comment]
I briefly implemented it: #4298
Please leave any comments.

**Comment by niboshi** [comment]
Fixed via #4559.
